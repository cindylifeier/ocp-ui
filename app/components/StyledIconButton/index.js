/**
 *
 * StyledIconButton
 *
 */
import styled from 'styled-components';
import IconButton from 'material-ui/IconButton';

const StyledIconButton = styled(IconButton).attrs({
  style: { position: 'relative' },
  iconStyle: {
    width: '100%',
    height: 26,
    position: 'absolute',
    top: '0',
    right: '0',
  },
})`
  & svg {
    fill: rgb(51, 51, 51) !important;
    transition: all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms;
  }

  &:hover svg {
    fill: #26a69a !important;
  }
`;

StyledIconButton.propTypes = {};

export default StyledIconButton;
