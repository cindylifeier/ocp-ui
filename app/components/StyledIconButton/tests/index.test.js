import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';
import { MuiThemeProvider } from 'material-ui';
import NavigationMenu from '@material-ui/icons/Menu';

import StyledIconButton from '../index';

configure({ adapter: new Adapter() });

describe('<StyledIconButton />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Act
      const renderedComponent = shallow(<StyledIconButton><NavigationMenu /></StyledIconButton>);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('style tests', () => {
    it('should have material-ui props for styling', () => {
      // Arrange
      const expectedProps = {
        style: { position: 'relative' },
        iconStyle: {
          width: '100%',
          height: 26,
          position: 'absolute',
          top: '0',
          right: '0',
        },
      };

      // Act
      const renderedComponent = shallow(<StyledIconButton><NavigationMenu /></StyledIconButton>);

      // Assert
      expect(renderedComponent.props().style).toEqual(expectedProps.style);
      expect(renderedComponent.props().iconStyle).toEqual(expectedProps.iconStyle);
    });

    it('should have styles from styled-components for svg descendant', () => {
      // Act
      const renderedComponent = mount(
        <MuiThemeProvider><StyledIconButton><NavigationMenu /></StyledIconButton></MuiThemeProvider>);

      // Assert
      expect(renderedComponent.find(StyledIconButton)).toHaveStyleRule('fill', 'rgb(51, 51, 51) !important', {
        modifier: ' svg',
      });
      expect(renderedComponent.find(StyledIconButton)).toHaveStyleRule('transition', 'all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms', {
        modifier: ' svg',
      });
    });

    it('should have styles from styled-components for svg descendant in hover state', () => {
      // Act
      const renderedComponent = mount(
        <MuiThemeProvider><StyledIconButton><NavigationMenu /></StyledIconButton></MuiThemeProvider>);

      // Assert
      expect(renderedComponent.find(StyledIconButton)).toHaveStyleRule('fill', '#26a69a !important', {
        modifier: ':hover svg',
      });
    });
  });
});
