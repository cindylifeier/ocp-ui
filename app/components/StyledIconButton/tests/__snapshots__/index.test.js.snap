// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StyledIconButton /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <StyledIconButton__StyledIconButton>
    <NavigationMenu />
  </StyledIconButton__StyledIconButton>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <NavigationMenu />,
      "className": "StyledIconButton__StyledIconButton-hHQfQP eubEyw",
      "disableTouchRipple": false,
      "disabled": false,
      "iconStyle": Object {
        "height": 26,
        "position": "absolute",
        "right": "0",
        "top": "0",
        "width": "100%",
      },
      "style": Object {
        "position": "relative",
      },
      "tooltipPosition": "bottom-center",
      "touch": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {},
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <NavigationMenu />,
        "className": "StyledIconButton__StyledIconButton-hHQfQP eubEyw",
        "disableTouchRipple": false,
        "disabled": false,
        "iconStyle": Object {
          "height": 26,
          "position": "absolute",
          "right": "0",
          "top": "0",
          "width": "100%",
        },
        "style": Object {
          "position": "relative",
        },
        "tooltipPosition": "bottom-center",
        "touch": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
